pytensor/link/pytorch/dispatch/basic.py

```diff
@@ -13,7 +13,12 @@
 @singledispatch
 def pytorch_typify(data, dtype=None, **kwargs):
     r"""Convert instances of PyTensor `Type`\s to PyTorch types."""
-    return torch.as_tensor(data, dtype=dtype)
+    try:
+        return torch.as_tensor(data, dtype=dtype)
+    except RuntimeError:
+        raise NotImplementedError(
+            f"pytorch_typify got type {type(data)}, it should be an array"
+        )
```

Comment:
This is not what I meant. The default should just raise NotImplementedError.

Then the old default of `torch.as_tensor` should be a function we dispatch for tensor/ndarray types, like we dispatch the None function on NoneType